appeals/web/src/server/appeals/appeal-details/withdrawal/withdrawal.mapper.js

```diff
@@ -0,0 +1,404 @@
+								document?.id &&
+								document?.latestDocumentVersion?.virusCheckStatus &&
+								document?.latestDocumentVersion?.virusCheckStatus === AVSCAN_STATUS.SCANNED
+									? `<a class="govuk-link" href="${mapDocumentDownloadUrl(
+											folder.caseId,
+											document.id
+									  )}">View</a>`.trim()
+									: ''
+						}
+					])
+				}
+			}
+		]
+	};
+
+	if (pageContent.pageComponents) {
+		preRenderPageComponents(pageContent.pageComponents);
+	}
+
+	return pageContent;
+}
+
+/**
+ *
+ * @param {Appeal} appealData
+ * @param {string} withdrawalRequestDay
+ * @param {string} withdrawalRequestMonth
+ * @param {string} withdrawalRequestYear
+ * @returns {PageContent}
+ */
+export function dateWithdrawalRequestPage(
+	appealData,
+	withdrawalRequestDay,
+	withdrawalRequestMonth,
+	withdrawalRequestYear
+) {
+	const title = 'Date of withdrawal request';
+
+	/** @type {PageComponent} */
+	const selectDateComponent = {
+		type: 'date-input',
+		parameters: {
+			id: 'withdrawal-request-date',
+			namePrefix: 'withdrawal-request-date',
+			fieldset: {
+				legend: {
+					text: 'Enter date',
+					classes: 'govuk-fieldset__legend--m'
+				}
+			},
+			hint: {
+				text: 'For example, 27 11 2023'
+			},
+			items: [
+				{
+					classes: 'govuk-input govuk-date-input__input govuk-input--width-2',
+					name: 'day',
+					value: withdrawalRequestDay || ''
+				},
+				{
+					classes: 'govuk-input govuk-date-input__input govuk-input--width-2',
+					name: 'month',
+					value: withdrawalRequestMonth || ''
+				},
+				{
+					classes: 'govuk-input govuk-date-input__input govuk-input--width-4',
+					name: 'year',
+					value: withdrawalRequestYear || ''
+				}
+			]
+		}
+	};
+
+	return {
+		title,
+		backLinkUrl: `/appeals-service/appeal-details/${appealData.appealId}/withdrawal/start`,
+		backLinkText: 'Back',
+		preHeading: `Appeal ${appealShortReference(appealData.appealReference)}`,
+		heading: title,
+		pageComponents: [
+			{
+				type: 'hint',
+				parameters: {
+					text: 'This is the date on the withdrawal correspondence from the appellant'
+				}
+			},
+			selectDateComponent
+		]
+	};
+}
+
+/**
+ *
+ * @param {Appeal} appealDetails
+ * @param {RedactionStatus[] | undefined} redactionStatuses
+ * @param {WithdrawalRequest} withdrawal
+ * @returns {PageContent}
+ */
+export function withdrawalDocumentRedactionStatusPage(
+	appealDetails,
+	redactionStatuses,
+	withdrawal
+) {
+	const redactionStatusesItems = redactionStatuses?.map((redactionStatus) => ({
+		value: redactionStatus.name.toLowerCase(),
+		text: redactionStatus.name,
+		checked: withdrawal?.redactionStatus === redactionStatus.name.toLowerCase()
+	}));
+
+	// if redaction status isn't set then pre-set it to 'unredacted' by default if available
+	if (!withdrawal?.redactionStatus && redactionStatusesItems) {
+		const redactionStatusesDefaultIndex = redactionStatusesItems?.findIndex(
+			(redactionStatusItem) => redactionStatusItem.value === 'unredacted'
```

Comment:
Redaction status values should use constants from `import { REDACTION_STATUS } from '@pins/appeals/constants/documents.js';` rather than hardcoding

Reply:
Not done. Explained in the chat